### PR TITLE
[REVIEW] minor conda recipe cleanup and bump librdkafka version

### PR DESCRIPTION
**Pull request by kkraus14**
Cleans up some of the dependency management in the conda recipes, notably removes `.*` from the ends of versions of packages, makes some header only dependencies `host` / `build` section only, and cleans up the import tests.

Additionally, upgrades `librdkafka` to 1.5 cc @jdye64 

Depends on: https://github.com/rapidsai/integration/pull/135

### CODE REVIEW

**Comment by jdye64** [comment]
@kkraus14 nice! thanks for letting me know.

**Comment by kkraus14** [comment]
rerun tests

**Comment by kkraus14** [comment]
@jakirkham any chance you could review this?

**Review by raydouglass — APPROVED** [review]

File: conda/recipes/libcudf/meta.yaml [hunk 1]
```diff
@@ -31,12 +31,10 @@ requirements:
     - cudatoolkit {{ cuda_version }}.*
     - arrow-cpp 1.0.1.*
     - arrow-cpp-proc * cuda
-    - flatbuffers
```
Comment (jakirkham) [inline conda/recipes/libcudf/meta.yaml:34]: Do we no longer need this?

File: conda/recipes/libcudf/meta.yaml [hunk 2]
```diff
@@ -31,12 +31,10 @@ requirements:
...
     - boost-cpp 1.72.0
     - dlpack
   run:
     - {{ pin_compatible('cudatoolkit', max_pin='x.x') }}
-    - dlpack
```
Comment (jakirkham) [inline conda/recipes/libcudf/meta.yaml:39]: This is just header only. So dropping this makes sense

**Review by jakirkham — COMMENTED** [review]
Looks good. Had a couple minor comments/questions for clarification

File: conda/recipes/libcudf/meta.yaml (hunk 1 above)
Comment (kkraus14) [reply conda/recipes/libcudf/meta.yaml:34]: This was needed to build Arrow from source which we no longer do in the context of a conda build.

**Review by jakirkham — APPROVED** [review]

**Comment by kkraus14** [comment]
Going to put in a PR into the integration repo bumping librdkafka there before merging.

**Comment by jakirkham** [comment]
Looks like the A100 CI failed due to HTTP issues. Would suggest we ignore it. This has been a bit of an ongoing issue and there's a good chance restarting just causes the same issue to show up on a different job.

**Comment by kkraus14** [comment]
> Looks like the A100 CI failed due to HTTP issues. Would suggest we ignore it. This has been a bit of an ongoing issue and there's a good chance restarting just causes the same issue to show up on a different job.

Yup, just waiting on the integration PR before merging.

**Comment by jakirkham** [comment]
Looks like [this job]( https://gpuci.gpuopenanalytics.com/blue/organizations/jenkins/rapidsai%2Fgpuci%2Fcudf%2Fprb%2Fcudf-gpu-build/detail/cudf-gpu-build/33872/pipeline ) and [this job]( https://gpuci.gpuopenanalytics.com/blue/organizations/jenkins/rapidsai%2Fgpuci%2Fcudf%2Fprb%2Fcudf-gpu-build/detail/cudf-gpu-build/33873/pipeline ) had some build issue. Trying to restart in case this is just some sporadic issue.

rerun tests

**Comment by jakirkham** [comment]
Thanks Keith! 😄

**Pull request merged**